tools/tasks.py

```diff
@@ -46,7 +46,7 @@ def _process_file(self, ifile, ofile, processor):
     def redact_file(self, name, ifile):
         ofile = os.path.join(self.target_dir, name)
         _, filename = os.path.split(name)
-        if ".gz" in filename:
+        if ".gz" in filename or "expvars.json" in filename:
```

Comment:
Now we have 2 exceptions to this, I feel it might be cleaner to avoid invoking the log redaction code at all. Especially given that `expvars.json` is not a binary file, but still does not need redaction applied. The warning isn't going to match the behaviour.

It gives us the opportunity to keep this class be purely responsible for redacting **_all_** files passed to it, without special cases.

It looks like we have a nice place to do it here:

```py
    def redact_and_zip(self, filename, log_type, salt, node, blacklist=[]):
        files = []
        redactor = LogRedactor(salt, self.tmpdir, blacklist)

        for name, fp in self.files.iteritems():
            files.append(redactor.redact_file(name, fp.name))

        prefix = "%s_%s_%s" % (log_type, node, self.start_time)
        self._zip_helper(prefix, filename, files)
```